src/chips/sam4l/dma.rs

```diff
@@ -212,6 +216,25 @@ impl DMAChannel {
 
         volatile_store(&mut registers.interrupt_enable, 1 << 1);
     }
+
+    pub fn do_xfer_buf(&self, pid: usize, 
+                       buf: Option<&'static mut [u8]>,
+                       len: usize) {
+        let ptr = buf.unwrap();
+        if len > ptr.len() {
```

Comment:
`len` doesn't seem to ever actually be used here, and it's simple to just _slice_ buf to a shorter length (e.g. `&buf[0..newlen]`).